uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/leave-request-popup-controller_test.js

```diff
@@ -0,0 +1,591 @@
+            $scope.$digest();
+          });
+
+          it('will calculate balance change', function () {
+            expect($ctrl.calculateBalanceChange).toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('calculate balance', function () {
+
+        describe('opening', function () {
+
+          it('has default', function () {
+            expect($ctrl.balance.opening).toEqual(0);
+          });
+
+        });
+
+        describe('change', function () {
+
+          it('has default', function () {
+            expect($ctrl.balance.change.amount).toEqual(0);
+          });
+
+          describe('when day type changed', function () {
+
+            describe('for single day', function () {
+
+              beforeEach(function () {
+                //select half_day_am  to get single day mock data
+                //$ctrl.uiOptions.multipleDays = false;
+                //$ctrl.uiOptions.fromDate = new Date();
+                $ctrl.uiOptions.selectedFromType = optionGroupMock.specificObject('hrleaveandabsences_leave_request_day_type', 'name', 'half_day_am');
+                $ctrl.calculateBalanceChange();
+                $scope.$digest();
+              });
+
+              it('will update balance', function () {
+                expect($ctrl.balance.change.amount).toEqual(-0.5);
+              });
+
+              it('will update closing balance', function () {
+                expect($ctrl.balance.closing).toEqual(-0.5);
+              });
+            });
+            describe('for multiple days', function () {
+
+              beforeEach(function () {
+                $ctrl.uiOptions.multipleDays = true;
+                //select all_day to get multiple day mock data
+                $ctrl.onDateChange(new Date(), true);
+                $ctrl.onDateChange(new Date(), false);
+                $ctrl.uiOptions.selectedFromType = optionGroupMock.specificObject('hrleaveandabsences_leave_request_day_type', 'name', 'all_day');
+                $ctrl.calculateBalanceChange();
+                $scope.$digest();
+              });
+
+              it('will update change amount', function () {
+                expect($ctrl.balance.change.amount).toEqual(-2);
+              });
+
+              it('will update closing balance', function () {
+                expect($ctrl.balance.closing).toEqual(-2);
+              });
+            });
+          });
+        });
+
+        describe('change details', function () {
+
+          describe('hidden', function () {
+
+            it('by default', function () {
+              expect($ctrl.uiOptions.isChangeExpanded).toBeFalsy();
+            });
+          });
+
+          describe('when expanded', function () {
+
+            beforeEach(function () {
+              $ctrl.uiOptions.isChangeExpanded = true;
+            });
+
+            it('will display balance breakdown', function () {
+              expect($ctrl.uiOptions.isChangeExpanded).toBeTruthy();
```

Comment:
Meaningless test, you have just changed manually the value of `$ctrl.uiOptions.isChangeExpanded`, what's the point testing it?